moveit_ros/moveit_servo/src/servo_calcs.cpp

```diff
@@ -811,18 +814,23 @@ bool ServoCalcs::enforcePositionLimits(sensor_msgs::msg::JointState& joint_state
             (joint_state.velocity.at(joint_idx) > 0 &&
              (joint_angle > (limits[0].max_position - parameters_->joint_limit_margin))))
         {
-          rclcpp::Clock& clock = *node_->get_clock();
-          RCLCPP_WARN_STREAM_THROTTLE(LOGGER, clock, ROS_LOG_THROTTLE_PERIOD,
-                                      node_->get_name()
-                                          << " " << joint->getName() << " close to a position limit. Halting.");
-          halting = true;
-          break;
+          joints_to_halt.push_back(joint);
         }
       }
     }
   }
-
-  return !halting;
+  if (!joints_to_halt.empty())
+  {
+    std::ostringstream joints_names;
+    auto joiner = std::experimental::make_ostream_joiner(joints_names, ", ");
+    std::transform(joints_to_halt.cbegin(), joints_to_halt.cend(), joiner,
+                   [](const auto& joint) { return joint->getName(); });
+    rclcpp::Clock& clock = *node_->get_clock();
+    RCLCPP_WARN_STREAM_THROTTLE(LOGGER, clock, ROS_LOG_THROTTLE_PERIOD,
+                                node_->get_name()
+                                    << " " << joints_names.str() << " close to a position limit. Halting.");
```

Comment:
I usually write this like below. Is there a specific reason you did it the other way?

```suggestion
    RCLCPP_WARN_STREAM_THROTTLE(LOGGER, node_->get_clock(), ROS_LOG_THROTTLE_PERIOD,
                                node_->get_name()
                                    << " " << joints_names.str() << " close to a position limit. Halting.");
```